--- Src/Base/AMReX_GpuDevice.H
+++ Src/Base/AMReX_GpuDevice.H
@@ -153,7 +153,7 @@
     // gfx10XX has 32 threads per wavefront else 64
     static AMREX_EXPORT constexpr int warp_size = __AMDGCN_WAVEFRONT_SIZE;
 #else
-    static AMREX_EXPORT constexpr int warp_size = AMREX_HIP_OR_CUDA_OR_SYCL(64,32,32);
+    static AMREX_EXPORT constexpr int warp_size = AMREX_HIP_OR_CUDA_OR_SYCL(64,32,64);
 #endif
 
     static unsigned int maxBlocksPerLaunch () noexcept { return max_blocks_per_launch; }
