--- Src/Base/AMReX_GpuDevice.H
+++ Src/Base/AMReX_GpuDevice.H
@@ -150,7 +150,7 @@
     // gfx10XX has 32 threads per wavefront else 64
     static constexpr int warp_size = __AMDGCN_WAVEFRONT_SIZE;
 #   else
-    static constexpr int warp_size = AMREX_HIP_OR_CUDA_OR_DPCPP(64,32,16);
+    static constexpr int warp_size = AMREX_HIP_OR_CUDA_OR_DPCPP(64,32,32);
 #   endif
 
     static unsigned int maxBlocksPerLaunch () noexcept { return max_blocks_per_launch; }
